File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
            unordered_set<int> numSet(nums.begin(), nums.end());
            int longest = 0;

            for (int num : numSet) {
                // Only start a sequence if num is its beginning
                if (numSet.find(num - 1) == numSet.end()) {
                    int current = num;
                    int streak  = 1;

                    while (numSet.find(current + 1) != numSet.end()) {
                        current++;
                        streak++;
                    }

                    longest = max(longest, streak);
                }
            }
            return longest;
    }
};

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    memo = []

    def helper(self, nums, i) -> int:
        if(i >= len(nums)):
            return 0

        if(self.memo[i] != -1):
            return self.memo[i]
        
        self.memo[i] = max(self.helper(nums, i+1), self.helper(nums, i+2) + nums[i])
        return self.memo[i]

    def rob(self, nums: List[int]) -> int:
        self.memo = [-1 for _ in range(len(nums))]
        return self.helper(nums, 0)
